lib/cylc/mp_pool.py

```diff
@@ -159,7 +163,14 @@ def process(self):
         for proc, ctx, callback, callback_args in self.runnings:
             ret_code = proc.poll()
             if ret_code is None:
-                runnings.append([proc, ctx, callback, callback_args])
+                if time.time() > ctx.init_time + self.cmd_timeout:
+                    os.killpg(proc.pid, SIGKILL)
```

Comment:
Should protect this with a `try ... except OSError` block, just in case the process exits normally between line 165 and this line.